src/components/code/code_block.tsx

```diff
@@ -232,28 +232,35 @@ export const EuiCodeBlock: FunctionComponent<EuiCodeBlockProps> = ({
     [preClasses, onKeyDown]
   );
 
-  const optionalStyles: CSSProperties = {};
-
-  if (overflowHeight) {
-    const property =
-      typeof overflowHeight === 'string' ? 'height' : 'maxHeight';
-    optionalStyles[property] = overflowHeight;
-  }
+  const optionalStyles: CSSProperties = useMemo(() => {
```

Comment:
also v optional but this naming is fairly unspecific and I was confused as to why the styles were getting applied to both the wrapping div and the `pre` child until I looked a bit closer. this cleanup can **definitely** wait until we convert EuiCodeBlock to emotion of course
```suggestion
  const overflowHeightStyles: CSSProperties = useMemo(() => {
```